src/lib/tls/asio/asio_stream.h

```diff
@@ -295,29 +291,21 @@ class Stream
        * This function is used to shut down SSL on the stream. The function call will block until SSL has been shut down
        * or an error occurs. Note that this will not close the lowest layer.
        *
+       * Note that this can be used in reaction of a received shutdown alert from the peer.
+       *
        * @param ec Set to indicate what error occured, if any.
        */
       void shutdown(boost::system::error_code& ec)
          {
-         try
+         try_with_error_code([&]
             {
             native_handle()->close();
-            }
-         catch(const TLS_Exception& e)
-            {
-            ec = e.type();
-            }
-         catch(const Botan::Exception& e)
-            {
-            ec = e.error_type();
-            }
-         catch(const std::exception&)
-            {
-            ec = Botan::ErrorType::Unknown;
-            }
+            }, ec);
+
+         send_pending_encrypted_data(ec);
```

Comment:
It's probably not the most consistent use of error code. But it's very convenient in a number of places, where we'd have to check the error code before proceeding otherwise.
I added a comment to the method.
Do you think we should change it?

Reply:
No, I think that's fine. I just pointed it out for further review passes to keep it off my head. The comment is helpful anyway. Thanks.